### PR TITLE
Output reading behavior, read all files even if some throws

### PR DESCRIPTION
**Pull request by nicolas-pierr**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
small tweak in contract of the output file reading

**What is the current behavior?** *(You can also link to an open issue here)*
The API reading order is not defined and if reading a file throws an `IOException`, the remaining files will not be read.

**What is the new behavior (if this is a feature change)?**
The API reading order is still not defined. But if reading a file throws an `IOException`, the remaining files will be read. And all the exceptions raised during the reading are throwed afterwards

### CODE REVIEW

File: ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java [hunk 1]
```diff
@@ -391,16 +391,18 @@ public AmplNetworkReader readMetrics(Map<String, String> metrics) throws IOExcep
                     continue;
                 }
                 List<String> tokens = parseExceptIfBetweenQuotes(trimedLine);
-                if (tokens.size() != 1 && tokens.size() != 2) {
-                    LOGGER.error("Wrong number of columns {} , expected 1 or 2: '{}'", tokens.size(), trimedLine);
+                if (!tokens.isEmpty() && tokens.size() != 1 && tokens.size() != 2) {
```
Comment (geofjamg) [inline ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java:394]: check could be done by just tokens.size() > 2 ?

File: ampl-executor/src/main/java/com/powsybl/ampl/executor/AmplModelExecutionHandler.java [hunk 2]
```diff
@@ -41,6 +46,8 @@
  */
 public class AmplModelExecutionHandler extends AbstractExecutionHandler<AmplResults> {
 
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(AmplModelExecutionHandler.class);
```
Comment (geofjamg) [inline ampl-executor/src/main/java/com/powsybl/ampl/executor/AmplModelExecutionHandler.java:49]: you could import Logger instead?

File: ampl-executor/src/main/java/com/powsybl/ampl/executor/AmplResults.java [hunk 3]
```diff
@@ -6,25 +6,28 @@
...
-        return new AmplResults(true);
-    }
-
     private final boolean success;
+    private final Map<String, String> indicators;
 
-    public AmplResults(boolean success) {
+    public AmplResults(boolean success, Map<String, String> indicators) {
         this.success = success;
+        this.indicators = indicators;
```
Comment (geofjamg) [inline ampl-executor/src/main/java/com/powsybl/ampl/executor/AmplResults.java:23]: We can add a Objects.requireNotNull here?

**Review by geofjamg — COMMENTED** [review]
@nicolas-pierr just a few minor comments

**Review by geofjamg — APPROVED** [review]

File: ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java [hunk 4]
```diff
@@ -79,7 +79,7 @@ private static AmplException createWrongNumberOfColumnException(int expected, in
         return new AmplException("Wrong number of columns " + actual + ", expected " + expected);
     }
 
-    private static List<String> parseExceptIfBetweenQuotes(String str) {
+    protected static List<String> parseExceptIfBetweenQuotes(String str) {
         List<String> tokens = new ArrayList<>();
         Matcher m = PATTERN.matcher(str);
         while (m.find()) {
```
Comment (flo-dup) [inline ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java:85]: As discussed, this while can be avoided with a better pattern. And all single quotes are replaced, but not double quotes. But it's ok for this PR!

**Review by flo-dup — APPROVED** [review]

File: ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java [hunk 5]
```diff
@@ -33,7 +33,7 @@ public class AmplNetworkReader {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AmplNetworkReader.class);
 
-    private static final Pattern PATTERN = Pattern.compile("([^']\\S*|'.+?')\\s*");
+    private static final Pattern PATTERN = Pattern.compile("([^'\\\"]\\S*|'.+?'|\\\".+?\\\")\\s*");
```
[the remark this answers is not in the document]
Comment (flo-dup) [reply ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java:36]: Just noting for a future PR, `^([^'\"]\S+)\s+(?>([^'\"\s]+)$|'(.+)'$|\"(.+)\"$)` should be better

**Pull request merged**